app/uk/gov/hmrc/agentsubscription/model/AgencyDetails.scala

```diff
@@ -60,12 +60,12 @@ object OverseasBusinessAddress {
     (__ \ "countryCode").read[String](overseasCountryCodeValidation))(OverseasBusinessAddress.apply _)
 }
 
-object BusinessDetails {
-  implicit val writes = Json.writes[BusinessDetails]
+object TradingDetails {
+  implicit val writes = Json.writes[TradingDetails]
 
-  implicit val reads: Reads[BusinessDetails] = (
+  implicit val reads: Reads[TradingDetails] = (
     (__ \ "tradingName").read[String](overseasNameValidation) and
-    (__ \ "businessAddress").read[OverseasBusinessAddress])(BusinessDetails.apply _)
+    (__ \ "tradingAddress").read[OverseasBusinessAddress])(TradingDetails.apply _)
 }
 
 object BusinessContactDetails {
```

Comment:
Down on line 85, the `(__ \ "agencyAddress").read[OverseasAgencyAddress]` is overseas specific, but the `AgencyDetails` is used for both UK and Overseas subscriptions (ie. `DesConnector.subscribeToAgentServices`).